torch/csrc/autograd/python_variable.cpp

```diff
@@ -1519,18 +1520,28 @@ void concrete_dispatch_fn(const c10::impl::PyInterpreter*, const c10::OperatorHa
     if (default_ivalue != ivalue) {
       break;
     }
-    default_suffix_len++;
   }
 
-  auto args = py::reinterpret_steal<py::object>(PyTuple_New(num_arguments - default_suffix_len));
+  // Find the split point between kwarg-only and regular.
+  // (Why +1?  As in the normal indexing convention, end is exclusive, begin is
+  // inclusive.  So if idx is actually kwarg only--we don't know yet--it is
+  // only included if I set end to +1)
+  const int64_t kwarg_only_end = idx + 1;
```

Comment:
I had to stare at this for a while haha. I think in a case like:
```
def foo(a, b=1, *, c=1, d=1):
    pass
foo(1, 1, 1, 1)
````
It's a little weird because kwarg_only_begin and kwarg_only_end won't actually point to indices for value keyword arguments (they'll both point to the index of `b`), but I guess that doesn't matter, since in any case we don't have to populate any kwargs since they're all defaulted.